src/components/RenderIfVisible/index.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useMemo, useRef, useState } from "react"
+
+type Props = {
+  initialVisible?: boolean
+  defaultHeight?: number
+  // How far outside the viewport in pixels should elements be considered visible?
+  visibleOffset?: number
+
+  root?: HTMLElement | null
+  rootElement?: string
+  rootElementClass?: string
+  children: React.ReactNode
+}
+
+const RenderIfVisible = ({
+  initialVisible = false,
+  defaultHeight = 300,
+  visibleOffset = 1000,
+  root = null,
+  rootElement = "div",
+  rootElementClass = "",
+  children,
+}: Props) => {
+  const [isVisible, setIsVisible] = useState<boolean>(initialVisible)
+  const intersectionRef = useRef<HTMLDivElement>(null)
+
+  // if height of the items are not equal
```

Comment:
commented section still needed?

Reply:
yes, let's keep it until the new design comes out.